### ISSUE TITLE
[BUG]: cookies->delete() does not work

### ISSUE THREAD

**Issue by wurst-hans** [phalcon/cphalcon#15405]
Using Phalcon 4.1.0 on PHP 7.4.16 there's an issue with deleting cookies.

```
$name = 'test-cookie';
var_dump($this->cookies->has($name));
var_dump($this->cookies->get($name)->getValue());
var_dump($this->cookies->delete($name));
var_dump($this->cookies->get($name)->delete());
return;
```

gives

```
bool(true)
string(43) "{"info":["test-info"],"warn":["test-warn"]}"
bool(false)
NULL
```

The only way to delete a cookie is, getting it from collection with `get($name)` and calling `delete()` on the cookie itself. Even when the last call returns `NULL`, the cookie is deleted then, but not when using `delete()` method of cookies collection. To be more precise, the required HTTP header for deleting a cookie: `Set-Cookie: test-cookie=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/` is added only when deleting directly on cookie object.
Why does `delete($name)` fails on `Phalcon\Http\Response\Cookies`. I cannot find any reason for this in source.

**Comment by niden** [comment]
@wurst-hans 

I am not sure how the cookies are populated in your app but here is what I see:

```
$this->cookies->has($name);
```
This one checks the internal cookie collection/arrray OR the `$_COOKIE` superglobal. My suspicion is that it returns `true` because that cookie exists in `$_COOKIE`

```
$this->cookies->get($name)->getValue();
```
It checks the internal collection. If found, it will return it back. If not found, it will pick up the cookie from the superglobal, create an object and then return it back to you. It will not store it in the internal collection because it will be sent twice otherwise.

```
$this->cookies->delete($name);
```
It checks the collection and if not there, it will return false. `delete` does not delete the cookie from the superglobal

```
$this->cookies->get($name)->delete();
```
This one is the same as above (`get`) but it should remove the cookie for you. It returns correctly `null` because the `delete()` method on `Cookie` does not return anything back. When you call the `delete()` on cookie it does a `setcookie()` call with a date way in the past. This is why you see that expiration date.

The code seems a bit convoluted for sure. I don't want to refactor it right now, since for v6 we will build a brand new HTTP layer to allow PSR-7 and all that. I checked the documentation and we do have a note next to the `delete` ([here - scroll a bit down](https://docs.phalcon.io/4.0/en/response#encryption)) but definitely needs a bit more information there as to what happens with the `get()` as I outlined it above.
